### PR TITLE
MINOR: Refine javadoc in TopicsDelta TopicDelta LocalReplicaChanges

### PR DESCRIPTION
**Pull request by brandboat**
Add more description to TopicsDelta TopicDelta LocalReplicaChanges

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/image/TopicDelta.java [hunk 1]
```diff
@@ -118,13 +118,15 @@ public boolean hasPartitionsWithAssignmentChanges() {
...
      * Find the partitions that have change based on the replica given.
-     *
+     * <p>
      * The changes identified are:
-     *   1. deletes: partitions for which the broker is not a replica anymore
-     *   2. electedLeaders: partitions for which the broker is now a leader (leader epoch bump on the leader)
-     *   3. leaders: partitions for which the isr or replicas change if the broker is a leader (partition epoch bump on the leader)
-     *   4. followers: partitions for which the broker is now a follower or follower with isr or replica updates (partition epoch bump on follower)
-     *
+     *   <ul>1. deletes: partitions for which the broker is not a replica anymore</ul>
```
Comment (showuon) [inline metadata/src/main/java/org/apache/kafka/image/TopicDelta.java]: Is it `<ul>` `<li>` ?

ref: https://blog.joda.org/2012/11/javadoc-coding-standards.html

File: metadata/src/main/java/org/apache/kafka/image/TopicsDelta.java [hunk 2]
```diff
@@ -178,13 +178,15 @@ public Set<Uuid> createdTopicIds() {
...
      * Find the topic partitions that have change based on the replica given.
-     *
+     * <p>
      * The changes identified are:
-     *   1. deletes: partitions for which the broker is not a replica anymore
-     *   2. electedLeaders: partitions for which the broker is now a leader (leader epoch bump on the leader)
-     *   3. leaders: partitions for which the isr or replicas change if the broker is a leader (partition epoch bump on the leader)
-     *   4. followers: partitions for which the broker is now a follower or follower with isr or replica updates (partition epoch bump on follower)
-     *
+     *   <ul>1. deletes: partitions for which the broker is not a replica anymore</ul>
```
Comment (showuon) [inline metadata/src/main/java/org/apache/kafka/image/TopicsDelta.java]: ditto

File: metadata/src/main/java/org/apache/kafka/image/TopicDelta.java [hunk 3]
```diff
@@ -118,13 +118,17 @@ public boolean hasPartitionsWithAssignmentChanges() {
...
+     *   <li>electedLeaders: partitions for which the broker is now a leader (leader epoch bump on the leader)</li>
+     *   <li>leaders: partitions for which the isr or replicas change if the broker is a leader (partition epoch bump on the leader)</li>
+     *   <li>followers: partitions for which the broker is now a follower or follower with isr or replica updates (partition epoch bump on follower)</li>
+     *   <li>topicIds: a map of topic names to topic IDs in leaders and followers changes</li>
+     *   <li>directoryIds: partitions for which directory id changes or newly added to the broker</li>
+     * </ul>
+     * <p>
      * Leader epoch bumps are a strict subset of all partition epoch bumps, so all partitions in electedLeaders will be in leaders.
      *
      * @param brokerId the broker id
```
Comment (showuon) [inline metadata/src/main/java/org/apache/kafka/image/TopicDelta.java:134]: Just found it, the `@return` should also update. Same as in `TopicsDelta.java`.

Comment (brandboat) [reply metadata/src/main/java/org/apache/kafka/image/TopicDelta.java:134]: Also modified the `@return` desc in the latest commit, thank you 😃

Comment (brandboat) [reply metadata/src/main/java/org/apache/kafka/image/TopicDelta.java:134]: gentle ping @showuon , could you take a look when you are available ? Thank you ! 😃

**Review by showuon — APPROVED** [review]
LGTM!

**Pull request merged**